### ISSUE TITLE
How to solve in case that the augmentation costs too much time, like 100s for a single wav file?

### ISSUE THREAD

**Issue by young169** [iver56/audiomentations#274]

**Comment by mmxgn** [comment]
Hi,

Thank you for using audiomentations.

Can we get more context? What augmentation are you trying to achieve, and how long is your file?

Some sample code would also be of great help.

Thanks again :)

**Comment by young169** [comment]
> Hi,
> 
> Thank you for using audiomentations.
> 
> Can we get more context? What augmentation are you trying to achieve, and how long is your file?
> 
> Some sample code would also be of great help.
> 
> Thanks again :)

I used the example code you presented, and just give it a real wav file input. 

I figured it out myself now: when I load it through soundfile, the augmentation runs in a unbearable speed, while turn to load it through librosa, the problem is gone. 
The main deference is that soundfile load a 2-dim ndarray while librosa's default setting returns a mono one.

I'm new to the audio domain so I wonder how much the 2-channel or mono affects the downstream tasks such as classification? Glad if you can share more on this!

**Comment by mmxgn** [comment]
Nice, I am glad you found it! Probably need to clarify this better in the documentation too from our side :) Feel free to close the issue if it is solved.

**Comment by iver56** [comment]
The ndarray shape expectation for multichannel audio is documented here: https://iver56.github.io/audiomentations/#multichannel-audio

If you give it some audio with the wrong shape, (e.g. `(44100, 2)` instead of `(2, 44100)`)  audiomentations will emit a warning:

```
warnings.warn(
    "Multichannel audio must have channels first, not channels last. In"
    " other words, the shape must be (channels, samples), not"
    " (samples, channels)"
)
```

Did you see this warning in the output from your program?

**Comment by iver56** [comment]
Should I make it an error instead of a warning?

**Comment by young169** [comment]
> 

Yes, I checked the console and found it, just didn't know what consequences it can cause that time.

**Comment by mmxgn** [comment]
> The ndarray shape expectation for multichannel audio is documented here: https://iver56.github.io/audiomentations/#multichannel-audio

This is kind of "hidden" imho, I would consider making it a bit more apparent in the usage example

**Comment by iver56** [comment]
I'd rather not make the code example more complicated than it already is. Instead, I will do the following:

* Write a guide: https://iver56.github.io/audiomentations/guides/multichannel_audio_array_shapes/
* Make it an error (with link guide) instead of a warning, forcing the user to fix the error before the code can run
